Docs/security assessment (#191)
Slug-Boi-ITU-Repositories/Dense-Dino-Nuggets: docs/Security_Assesment.md

| User credentials | External attackers | Weak password hashing allows attackers to crack stolen hashes and reuse credentials |

## Risk Analysis

Below we define the scale for our risk matrix for our analysis. Then we analyse the risk of the scenarios and describe possible ways of mitigating risk.

### Likelihood Scale

| Level | Description |
|-------|-------------|
| Low (1) | Difficult to exploit, requires special conditions |
| Medium (2) | Possible with some effort |
| High (3) | Easy to exploit, common attack |

### Impact Scale

| Level | Description |
|-------|-------------|
| Low (1) | Minor impact, no sensitive data |
| Medium (2) | Partial data exposure or service disruption |
| High (3) | Major data breach or full system compromise |

### Risk Matrix

Risk Score = Likelihood × Impact

| Score | Risk Level |
|-------|------------|
| 1–2 | Low |
| 3–4 | Medium |
| 6 | High |
| 9 | Critical |

### Risk Evaluation and Mitigation

| Asset | Scenario | Likelihood | Impact | Score | Risk Level | Mitigation Strategy |
|-------|----------|------------|--------|-------|------------|---------------------|

> | Backend API | Injection vulnerability | 1 | 3 | 9 | Critical | Use GORM (ORM) with parameterized queries, input validation, and escaping html input |

Reviewer: ```suggestion
| Backend API | Injection vulnerability | 1 | 3 | 3 | Medium | Use GORM (ORM) with parameterized queries, input validation, and escaping html input |
```

| PostgreSQL database | Credential compromise | 2 | 3 | 6 | High | Restrict database access to internal network, use strong credentials, do not expose database publicly |
| Session tokens | Token theft | 2 | 3 | 6 | High | Use secure cookies, HTTPS, short expiry |
| Docker Swarm | Cluster takeover | 2 | 3 | 6 | High | Do not expose Swarm manager, restrict access via firewall, secure configuration |
| Monitoring system | Exposure of information | 2 | 2 | 4 | Medium | Require authentication for dashboards, limit access |

> | DigitalOcean droplets | SSH compromise | 1 | 3 | 6 | High | Use SSH key-based authentication, apply firewall restrictions |

Reviewer: ```suggestion
| DigitalOcean droplets | SSH compromise | 1 | 3 | 3 | Medium | Use SSH key-based authentication, apply firewall restrictions |
```

> | Secrets/config | Credential leak | 1 | 3 | 9 | Critical | Store secrets securely - "Secrets belpng in vaults", Use enviroment variables |

Reviewer: ```suggestion
| Secrets/config | Credential leak | 1 | 3 | 3 | Medium | Store secrets securely - "Secrets belong in vaults", Use environment variables |
```

> | Ingress | Service exposure | 2 | 2 | 4 | High | Stop exposing ports that should not be exposed publically |

Reviewer: ```suggestion
| Ingress | Service exposure | 2 | 2 | 4 | Medium | Stop exposing ports that should not be exposed publically |
```

> | User credentials | Weak hashing | 1 | 3 | 6 | High | Use strong hashing + salting |

Reviewer: ```suggestion
| User credentials | Weak hashing | 1 | 3 | 3 | Medium | Use strong hashing + salting |
```
